twofactor_privacyidea/lib/Provider/TwoFactorPrivacyIDEAProvider.php

```diff
@@ -615,46 +633,61 @@ private function processPIResponse(IResponse $result)
                                     case "u2f":
                                         $this->session->set("pi_u2fSignRequest", $multiChallenge[$i]->attributes->u2fSignRequest);
                                         $triggeredTokenTypes[] = "u2f";
+                                        if (!empty($multiChallenge[$i]->image) && !empty($multiChallenge[$i]->client_mode) && $multiChallenge[$i]->client_mode === "u2f")
```

Comment:
I know it is not well looking. I've made it so because it has to be refactored anyway when we remove the preferredTokenType option. But I think we should consider removing the preferredTokenType now because this option wasn't released anyway. And then I can refactor this part of code. What do you think about it?

Reply:
if `preferredTokenType` is not in the currently released version, you can remove it now, yes

Reply:
ok